mulog-filesystem-metrics/src/com/brunobonacci/mulog/publishers/filesystem_metrics.clj

```diff
@@ -0,0 +1,103 @@
+(ns com.brunobonacci.mulog.publishers.filesystem-metrics
+  (:require [clojure.spec.alpha :as s]
+            [com.brunobonacci.mulog :as u]
+            [com.brunobonacci.mulog.buffer :as rb])
+  (:import  [java.nio.file FileSystems FileSystem FileStore]))
+
+
+
+(def store-hacks
+  [(when-let [klazz (try (Class/forName "sun.nio.fs.UnixFileStore")
+                         (catch ClassNotFoundException _ nil))]
+     (let [field (doto (.getDeclaredField ^Class klazz "file")
+                   (.setAccessible true))]
+       [klazz (fn unix-file-store-hack [store] (str (.get field store)))]))
+
+   (when-let [klazz (try (Class/forName "sun.nio.fs.WindowsFileStore")
+                         (catch ClassNotFoundException _ nil))]
+     (let [field (doto (.getDeclaredField ^Class klazz "root")
+                   (.setAccessible true))]
+       [klazz (fn windows-file-store-hack [store] (str (.get field store)))]))])
```

Comment:
This is generating a massive compilation warning
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.brunobonacci.mulog.publishers.filesystem_metrics$fn__790 to field sun.nio.fs.UnixFileStore.file
WARNING: Please consider reporting this to the maintainers of com.brunobonacci.mulog.publishers.filesystem_metrics$fn__790
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

could you please add a comment explaining what this `store-hack` does and why it is necessary?

Reply:
This is to get the mount path of each volume (`FileStore`). Unfortunately, there doesn't seem to be a supported way to get at the, despite it always being there in some form, and it even appearing in the string representation of the object (see [this](https://stackoverflow.com/questions/10678363/find-the-directory-for-a-filestore) StackOverflow question).

The options as far as I can see are either:
- Live with it (and add a comment explaining what's going on). However, this may stop working completely on later versions of Java.
- Parse the string representation of the `FileStore` to extract the path (this might be as simple as splitting on space and taking one part), but I'm not sure how stable the representation is across platforms and Java versions.
- Get the information from some other source, for example shelling out to `mount`. This is likely to be even more system dependent than parsing the string representation of the `FileStore`.
- Remove the path from the metrics. However, this is what most people use to recognise a volume, so I think the metrics would be much harder to interpret without the mount path.

What do you think?

Reply:
I know it is quite disappointing.....

I think we can just add a comment explaining why we are doing this, referencing anything you can to support the decision. And if it stops working then we can adopt the more lame approach of parsing the `toString()` output.

Reply:
Added a comment, is that OK?

Reply:
Perfect thanks